ex03.ex

```diff
@@ -55,7 +55,40 @@ defmodule Ex03 do
 
   """
 
-  def odd_even . . . "your code"
+    def reverse([]) do [] end
+    def reverse([h|t]) do reverse(t) ++ [h] end
+
```

Comment:
## -4

So, it works... but there's so much duplication! And so much complication.

How about:

~~~ elixir
  def odd_even([]), do: []
  def odd_even([ h | t ]), do: [ atom_for(h) | odd_even(t) ]

  defp atom_for(n) when Integer.is_odd(n),  do:  :odd
  defp atom_for(n) when Integer.is_even(n), do:  :even
~~~

In general, if you find yourself writing identical code for the first case and the rest of the cases, you pretty much know there's going to be a simpler way...